muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/workspaces/index.ts

```diff
@@ -762,49 +857,232 @@ let updateWorkspace:UpdateWorkspaceTriggerType = function updateWorkspace(data){
+        }
+      });
+
+      let update: WorkspaceUpdateType = {
+        staffMemberSelect: { users: staffMemberSelect, state: "READY" },
+        id: workspace.id
+      }
+
+      dispatch({
+        type: 'UPDATE_ORGANIZATION_SELECTED_WORKSPACE',
+        payload: update
+      });
+    }
+    catch (err) {
+      if (!(err instanceof MApiError)) {
+        throw err;
+      }
+      dispatch(displayNotification(getState().i18n.text.get('plugin.organization.workspaces.notification.selectStaff.error'), 'error'));
+      dispatch({
+        type: 'UPDATE_ORGANIZATION_SELECTED_WORKSPACE',
+        payload: { staffMemberSelect: { state: "ERROR" } }
+      });
+    }
+  }
+}
+
+let loadStaffMembersOfWorkspace: LoadStaffMembersOfWorkspaceTriggerType = function loadStaffMembersOfWorkspace(workspace) {
+  return async (dispatch: (arg: AnyActionType) => any, getState: () => StateType) => {
     try {
       let staffMembers = <Array<UserStaffType>>(await promisify(mApi().user.staffMembers.read({
         workspaceEntityId: workspace.id,
         properties: 'profile-phone,profile-vacation-start,profile-vacation-end'
       }), 'callback')());
 
+      let update: WorkspaceUpdateType = {
+        staffMembers
+      }
+
       dispatch({
         type: 'UPDATE_WORKSPACE',
         payload: {
           original: workspace,
-          update: {
-            staffMembers
-          }
+          update
         }
       });
-    } catch (err){
-      if (!(err instanceof MApiError)){
+    } catch (err) {
+      if (!(err instanceof MApiError)) {
         throw err;
       }
-      dispatch(displayNotification(getState().i18n.text.get('TODO ERRORMSG failed to load teachers'), 'error'));
+      dispatch(displayNotification(getState().i18n.text.get('plugin.organization.workspaces.notification.selectStaff.error'), 'error'));
     }
   }
 }
 
-let loadStudentsOfWorkspace:LoadStudentsOfWorkspaceTriggerType = function loadStudentsOfWorkspace(workspace){
-  return async (dispatch:(arg:AnyActionType)=>any, getState:()=>StateType)=>{
+let loadCurrentOrganizationWorkspaceSelectStudents: LoadStudentsOfWorkspaceTriggerType = function loadCurrentOrganizationWorkspaceSelectStudents(workspace) {
+  return async (dispatch: (arg: AnyActionType) => any, getState: () => StateType) => {
     try {
-      let students = <Array<ShortWorkspaceUserWithActiveStatusType>>(await promisify(mApi().workspace.workspaces.students.read(workspace.id), 'callback')());
 
-      let update:WorkspaceUpdateType = {
-          students
+      dispatch({
+        type: 'UPDATE_ORGANIZATION_SELECTED_WORKSPACE',
+        payload: { id: workspace.id, studentsSelect: { state: "LOADING", users: [] } }
+      });
+
+      let students: ShortWorkspaceUserWithActiveStatusType[] = <Array<ShortWorkspaceUserWithActiveStatusType>>(await promisify(mApi().workspace.workspaces.students.read(workspace.id), 'callback')());
+
+      let selectStudents: SelectItem[] = students.map((student) => {
```

Comment:
Usage of select item once again, it shouldn't be here, why are we defining an icon here?

Reply:
This comment points to a wrong line number and therefore cannot automatically be resolved as it will never go outdated, there fore resolving this by hand manually.